simple/stats/data.py

```diff
@@ -181,6 +187,81 @@ class Observation:
   provenance: str
 
 
+@dataclass
+class Property:
+  dcid: str
+  name: str
+
+  def triples(self) -> list[Triple]:
+    return [
+        Triple(self.dcid, _PREDICATE_TYPE_OF, object_id=_PROPERTY),
+        Triple(self.dcid, _PREDICATE_NAME, object_value=self.name),
+    ]
+
+
+@dataclass
+class EventType:
+  id: str
+  name: str
+  description: str = ""
+  provenance_ids: list[str] = field(default_factory=list)
+  source_ids: list[str] = field(default_factory=list)
+
+  def add_provenance(self, provenance: "Provenance") -> "EventType":
+    provenance_id = provenance.id
+    source_id = provenance.source_id
+    if not provenance_id in self.provenance_ids:
+      self.provenance_ids.append(provenance_id)
+    if not source_id in self.source_ids:
+      self.source_ids.append(source_id)
+
+    return self
+
+  def triples(self) -> list[Triple]:
+    triples: list[Triple] = []
+    triples.append(Triple(self.id, _PREDICATE_TYPE_OF, object_id=_CLASS))
+    triples.append(Triple(self.id, _PREDICATE_SUB_CLASS_OF, object_id=_EVENT))
+    triples.append(Triple(self.id, _PREDICATE_NAME, object_value=self.name))
+    if self.description:
+      triples.append(
+          Triple(self.id, _PREDICATE_DESCRIPTION,
+                 object_value=self.description))
+    for provenance_id in self.provenance_ids:
+      triples.append(
+          Triple(self.id, _PREDICATE_INCLUDED_IN, object_id=provenance_id))
+    for source_id in self.source_ids:
+      triples.append(
+          Triple(self.id, _PREDICATE_INCLUDED_IN, object_id=source_id))
+    return triples
+
+
+@dataclass
+class Event:
+  id: str
+  event_type: str
+  entity_type: str = ""
+  entity: str = ""
+  date: str = ""
+  provenance_id: str = ""
+  properties: dict[str, str] = field(default_factory=lambda: defaultdict(dict))
+
+  def triples(self) -> list[Triple]:
+    triples: list[Triple] = []
+    triples.append(
+        Triple(self.id, _PREDICATE_TYPE_OF, object_id=self.event_type))
+    if self.entity and self.entity_type:
+      triples.append(Triple(self.id, self.entity_type, object_id=self.entity))
```

Comment:
the predicate should not be "self.entity_type"?

Reply:
Thanks - changed to `location`.